go/cmd/dolt/commands/reset.go

```diff
@@ -254,15 +254,28 @@ func printNotStaged(ctx context.Context, dEnv *env.DoltEnv, staged *doltdb.RootV
 		return
 	}
 
-	if notStagedTbls.NumRemoved+notStagedTbls.NumModified+notStagedDocs.NumRemoved+notStagedDocs.NumModified > 0 {
-		cli.Println("Unstaged changes after reset:")
+	removeModified := 0
+	for _, td := range notStagedTbls {
+		if !td.IsAdd() {
+			removeModified++
+		}
+	}
 
-		lines := make([]string, 0, notStagedTbls.Len()+notStagedDocs.Len())
-		for _, tblName := range notStagedTbls.Tables {
-			tdt := notStagedTbls.TableToType[tblName]
+	if removeModified+notStagedDocs.NumRemoved+notStagedDocs.NumModified > 0 {
+		cli.Println("Unstaged changes after reset:")
 
-			if tdt != diff.AddedTable && !doltdb.IsReadOnlySystemTable(tblName) {
-				lines = append(lines, fmt.Sprintf("%s\t%s", tblDiffTypeToShortLabel[tdt], tblName))
+		var lines []string
+		for _, td := range notStagedTbls {
+			if td.IsAdd() {
```

Comment:
This logic is repeated nearly verbatim in several places. Can you extract a method?

Reply:
I plan on cleaning up the printing logic, but I have a larger refactor involving DocDiffs planned. 
For now, it just does the right thing